worker/upgradedatabase/manifold.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package upgradedatabase
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/worker/v3"
+	"github.com/juju/worker/v3/dependency"
+
+	"github.com/juju/juju/worker/gate"
+)
+
+// Logger defines the logging methods used by the worker.
+type Logger interface {
+	Criticalf(string, ...interface{})
+	Errorf(string, ...interface{})
+	Warningf(string, ...interface{})
+	Infof(string, ...interface{})
+	Debugf(string, ...interface{})
+	Tracef(string, ...interface{})
+}
```

Comment:
Do we need all of these? 

Also can you change `interface{}` to `any`.

Reply:
I've removed Critical and Trace (all others look like used in prev worker mongoDB-version)